### PR TITLE
BaSP-TG-127: Textarea shared component

### PR DESCRIPTION
**Pull request by MaicolFreire001**
Created a text area shared component and replaced all text area already existing with it

### CODE REVIEW

File: src/Components/Shared/TextArea/textArea.module.css [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* .formcontrol {
```
Comment (Ignacio-Sormani) [inline src/Components/Shared/TextArea/textArea.module.css:1]: Remove comments

File: src/Components/Shared/TextArea/index.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const TextAreaField = ({ label, name, placeholder, register, columns, rows, error = '' }) => {
+  return (
+    <div>
+      {label && <label>{label}</label>}
+      <textarea
+        name={name}
+        placeholder={placeholder}
+        {...register(name)}
+        cols={columns}
+        rows={rows}
```
Comment (Ignacio-Sormani) [inline src/Components/Shared/TextArea/index.jsx:12]: You are not passing values for row when you use this component. So for textarea element, rows is undefined. Maybe add a default value por rows

File: src/Components/Shared/TextArea/textArea.module.css (hunk 1 above)
Comment (MaicolFreire001) [reply src/Components/Shared/TextArea/textArea.module.css:1]: Done

File: src/Components/Shared/TextArea/index.jsx (hunk 2 above)
Comment (MaicolFreire001) [reply src/Components/Shared/TextArea/index.jsx:12]: Removed rows prop altogether as it was doing nothing

**Comment by fede230298** [comment]
TextArea should be used in Task too. Please Add

File: src/Components/Projects/Form/index.js [hunk 3]
```diff
@@ -171,13 +172,13 @@ const Project = () => {
...
-            {...register('description')}
+          <TextAreaField
             className={styles.textarea}
-          ></textarea>
+            name="description"
+            placeholder="Description"
+            register={register}
+            columns="100"
+          />
           {errors.description && <p>{errors.description.message}</p>}
```
Comment (ProDIGGY-vinilos) [inline src/Components/Projects/Form/index.js:182]: this error should be delete or i am wrong ?

File: src/Components/TimeSheets/Form/index.js [hunk 4]
```diff
@@ -170,11 +171,12 @@ const TimeSheetsForm = () => {
...
-          <textarea
+          <TextAreaField
             name="description"
-            {...register('description')}
             placeholder="Description"
-          ></textarea>
+            register={register}
+            columns={30}
+          />
           {errors.description && <p>{errors.description.message}</p>}
```
Comment (ProDIGGY-vinilos) [inline src/Components/TimeSheets/Form/index.js:180]: same here

File: src/Components/Pages/Employee/TimeSheetsList/Form/index.jsx [hunk 5]
```diff
@@ -116,12 +117,12 @@ const TimeSheetsForm = () => {
...
-            className={styles.textArea}
+          <TextAreaField
             name="description"
-            {...register('description')}
             placeholder="Description"
-          ></textarea>
+            register={register}
+            columns={40}
+          />
           {errors.description && <p>{errors.description.message}</p>}
```
Comment (ProDIGGY-vinilos) [inline src/Components/Pages/Employee/TimeSheetsList/Form/index.jsx:126]: and same here

**Review by JuliMonte5 — COMMENTED** [review]
Implement the component in tasks input too

File: src/Components/Projects/Form/index.js (hunk 3 above)
Comment (MaicolFreire001) [reply src/Components/Projects/Form/index.js:182]: Done

File: src/Components/TimeSheets/Form/index.js (hunk 4 above)
Comment (MaicolFreire001) [reply src/Components/TimeSheets/Form/index.js:180]: Done

File: src/Components/Pages/Employee/TimeSheetsList/Form/index.jsx (hunk 5 above)
Comment (MaicolFreire001) [reply src/Components/Pages/Employee/TimeSheetsList/Form/index.jsx:126]: Done

**Review by JuliMonte5 — APPROVED** [review]

**Review by fede230298 — APPROVED** [review]

**Review by sofihoj — CHANGES_REQUESTED** [review]
You could make the textarea have a fixed size instead of being able to enlarge it manually

**Review by sofihoj — APPROVED** [review]

**Pull request merged**